src/utils/source.js

```diff
@@ -329,3 +340,16 @@ export function getTextAtPosition(source: Source, location: Location) {
 
   return lineText.slice(column, column + 100).trim();
 }
+
+export function getSourceClassnames(
+  source: Object,
+  ...additionalClasses: Array<string>
+) {
+  const sourceRecord = new SourceRecordClass(source);
+  if (source && sourceRecord.get("isBlackBoxed")) {
```

Comment:
this seems like it is doing too much...

It is probably sufficient to return blackbox or the sourceType